homeassistant/components/sensor/bme680.py

```diff
@@ -0,0 +1,373 @@
+            self.temperature = None
+            self.humidity = None
+            self.pressure = None
+            self.gas_resistance = None
+            self.air_quality = None
+
+    def __init__(
+            self, sensor, gas_measurement=False,
+            burn_in_time=300, hum_baseline=40, hum_weighting=25
+    ):
+        """Initialize the sensor handler."""
+        self.sensor_data = BME680Handler.SensorData()
+        self._sensor = sensor
+        self._gas_sensor_running = False
+        self._hum_baseline = hum_baseline
+        self._hum_weighting = hum_weighting
+        self._gas_baseline = None
+
+        if gas_measurement:
+            import threading
+            threading.Thread(
+                target=self._run_gas_sensor,
+                kwargs={'burn_in_time': burn_in_time},
+                name='BME680Handler_run_gas_sensor'
+            ).start()
+        self.update(first_read=True)
+
+    def _run_gas_sensor(self, burn_in_time):
+        """Calibrate the Air Quality Gas Baseline."""
+        if not self._gas_sensor_running:
+            self._gas_sensor_running = True
+            import time
+
+            # Pause to allow inital data read for device validation.
+            time.sleep(1)
+
+            start_time = time.time()
+            curr_time = time.time()
+            burn_in_data = []
+
+            _LOGGER.info(("Beginning %d second gas sensor burn in for "
+                          "Air Quality"), burn_in_time)
+            while curr_time - start_time < burn_in_time:
+                curr_time = time.time()
+                if (
+                        self._sensor.get_sensor_data() and
+                        self._sensor.data.heat_stable
+                ):
+                    gas_resistance = self._sensor.data.gas_resistance
+                    burn_in_data.append(gas_resistance)
+                    self.sensor_data.gas_resistance = gas_resistance
+                    _LOGGER.debug(("AQ Gas Resistance Baseline reading %2f "
+                                   "Ohms"), gas_resistance)
+                    time.sleep(1)
+
+            _LOGGER.debug(("AQ Gas Resistance Burn In Data (Size: %d): "
+                           "\n\t%s"), len(burn_in_data), burn_in_data)
+            self._gas_baseline = sum(burn_in_data[-50:]) / 50.0
+            _LOGGER.info("Completed gas sensor burn in for Air Quality")
+            _LOGGER.info("AQ Gas Resistance Baseline: %f", self._gas_baseline)
+            while True:
+                if (
+                        self._sensor.get_sensor_data() and
+                        self._sensor.data.heat_stable
+                ):
+                    self.sensor_data.gas_resistance = (
+                        self._sensor.data.gas_resistance
+                    )
+                    self.sensor_data.air_quality = self._calculate_aq_score()
+                    time.sleep(1)
+        else:
+            return
```

Comment:
You could move this to the initial check `if not self._gas_sensor_running` after inverting that. Then you could outdent most of the lines in this method.